### DISCUSSION TITLE
Regarding third column of ruff --statistics

### DISCUSSION THREAD

**Discussion by jorgegomezcq** [astral-sh/ruff#8392] [Q&A]
Hi,

What does the third column in the below `ruff --statistics .` output signify?

I'm guessing that the `[*]` might mean that it was able to auto-fix everything and `[]` means that it was not able to auto-fix everything.

But, I'm not sure since there are no descriptive column headers. 

My next step will be to try reading the source code to see what Rust `struct` fields these columns might come from...

WDYT 

```sh
$ ruff --statistics .
73	D212	[*] Multi-line docstring summary should start at the first line
21	D202	[*] No blank lines allowed after function docstring (found 1)
10	D403	[*] First word of the first line should be capitalized: `foo` -> `Foo`
 5	D200	[*] One-line docstring should fit on one line
 3	D405	[*] Section name should be properly capitalized ("TODO")
 3	D411	[*] Missing blank line before section ("Args")
 3	D416	[*] Section name should end with a colon ("Args")
 2	D209	[*] Multi-line docstring closing quotes should be on a separate line
 2	D419	[ ] Docstring is empty
 1	D210	[ ] No whitespaces allowed surrounding docstring text
 1	D412	[*] No blank lines allowed between a section header and its content ("Note")
```

**Comment by jorgegomezcq** [comment]
It looks like it comes from `crates/ruff_cli/src/printer.rs`:

```rust
   ...
   let fixable = format!("[{}] ", "*".cyan());
   let unfixable = "[ ] ";
```

**Reply by charliermarsh** [reply]
It's meant to indicate whether the rule is fixable. It matches the output of `ruff check`, which uses that glyph to indicate the number of fixable diagnostics at the bottom.

**Reply by jorgegomezcq** [reply]
Hi,

Thanks for the reply.

Should we add a legend for what the glyph means somewhere for the unsuspecting new ruff CLI user?

One idea I have without fully understanding the context around that ascii table API yet is to add a header to that ascii table describing what each column is.

**Reply by charliermarsh** [reply]
Yeah, I'm open to adding a header!

**Reply by jorgegomezcq** [reply]
Cool! I'll try to work on this soon. Thanks for the reply.
